bluemira/base/reactor_config.py

```diff
@@ -229,12 +226,51 @@ def _check_args_are_strings(self, args: Iterable[str]):
             if not isinstance(a, str):
                 raise ReactorConfigError("args must be strings")
 
+    def _expand_paths_in_dict(self, d: Dict[str, Any], rel_path: Path):
+        """
+        Expand all file paths in the dict
+        by replacing their values with the json file contents.
+
+        This mutates the passed in dict.
+        """
+        for k in d:
+            d[k], rel_path_from = self._extract_and_expand_file_data_if_needed(
+                d[k], rel_path
+            )
+            if isinstance(d[k], dict):
+                self._expand_paths_in_dict(d[k], rel_path_from)
+
+    def _extract_and_expand_file_data_if_needed(
+        self, value: Any, rel_path: Path
+    ) -> Tuple[Union[Any, dict], str]:
+        if not isinstance(value, str):
+            return value, rel_path
+        if not value.startswith(_FILEPATH_PREFIX):
+            return value, rel_path
+
+        # remove _FILEPATH_PREFIX
+        f_path = value[len(_FILEPATH_PREFIX) :]
+
+        # check if rel path
+        if not f_path.startswith("/"):
+            f_path = rel_path / f_path
+        else:
+            f_path = Path(f_path)
+
+        # check if file exists
+        if not f_path.is_file():
+            raise FileNotFoundError(f"Cannot find file {f_path}")
+
+        f_data = self._read_json_file(f_path.as_posix())
```

Comment:
Not sure that we need the `as_posix` here, `open` can handle `pathlib.Path` objects - although maybe you were just trying to keep the type-checker happy with what `_read_json_file` expects. Fine either way, but thought it was worth mentioning.

Reply:
Ya def don't need it!